src/defiProvider.ts

```diff
@@ -120,6 +121,17 @@ export class DefiProvider {
         blockNumber,
       })
 
+      await resolveUnderlyings(
+        adapter,
+        blockNumber,
+        protocolPositions,
+        (underlyingToken, protocolToken, underlyingRateRaw) => {
+          underlyingToken.balanceRaw =
+            (protocolToken.balanceRaw * underlyingRateRaw) /
+            10n ** BigInt(protocolToken.decimals)
```

Comment:
This method is different depending on the field that needs being added (`balanceRaw` for positions and movements, and `totalValueRaw` for TVL).

One possible alternative here, since the calculation seems to be always the same, is to pass instead the field name that needs to be updated as a string (so it can be accessed as an index). I'll see how that looks, but it can cause more type problems.